src/WebApp.Web/Base/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,49 @@
+using System.Text.Json;
+
+namespace WebApp.Web.Base;
+
+public class ExceptionHandlingMiddleware: IMiddleware
+{
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        try
+        {
+            await next(context);
+        }
+        catch (CustomValidationException e)
+        {
+            await HandleCustomValidationException(context, e);
+        }
+        catch (Exception e)
+        {
+            await HandleExceptionAsync(context, e);
+        }
+    }
+
+    private async Task HandleCustomValidationException(HttpContext httpContext, CustomValidationException exception)
+    {
+        var response = new
```

Comment:
https://learn.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-5.0#default-badrequest-response-2 type and traceId are missing

Reply:
done